include/boost/math/tools/lexical_cast.hpp

```diff
@@ -17,11 +20,54 @@
 #endif
 
 namespace boost {
-    template <typename T1, typename T2>
-    inline T1 lexical_cast(const T2)
+
+    template <typename T, typename std::enable_if<std::is_same<T, float>::value, bool>::type = true>
+    inline T lexical_cast(const char* s)
```

Comment:
Are there ever any situations where we are lexical-casting to a float?  That would seem to be a mistake on our part if so?

Reply:
There were a couple of these that I remember hitting in the CI so I added all of the trivial facilities in case there was a hidden case or a need down the line for some reason.